### PR TITLE
PE-145:[NotifD] - migrate notifications.xml to the new CM

### PR DESCRIPTION
**Pull request by uguggilam**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/PE-145

### CODE REVIEW

File: opennms-config/src/main/java/org/opennms/netmgt/config/NotifdConfigFactory.java [hunk 1]
```diff
@@ -86,16 +91,17 @@ static synchronized public NotifdConfigFactory getInstance() {
...
      * <p>init</p>
      *
      * @throws java.io.IOException if any.
      * @throws java.io.FileNotFoundException if any.
      */
-    public static synchronized void init() throws IOException, FileNotFoundException {
+    public static synchronized void init() throws IOException {
         if (!initialized) {
             instance = new NotifdConfigFactory();
-            instance.reload();
```
Comment (freddy61025) [inline opennms-config/src/main/java/org/opennms/netmgt/config/NotifdConfigFactory.java:98]: you better leave a comment state where the reload go.

File: opennms-webapp/src/main/webapp/admin/sched-outages/index.jsp [hunk 2]
```diff
@@ -70,6 +70,7 @@
 			WriteablePollOutagesDao.class);
 
 	NotifdConfigFactory.init(); //Must do this early on - if it fails, then just throw the exception to the web gui
+    NotifdConfigFactory.getInstance().reload();
```
Comment (freddy61025) [inline opennms-webapp/src/main/webapp/admin/sched-outages/index.jsp:73]: is that possible to not call reload here ? (your init reload is removed, and postConstruct did not work properly?)

File: opennms-webapp/src/main/webapp/admin/sched-outages/editoutage.jsp [hunk 3]
```diff
@@ -158,7 +158,8 @@
 		}
 	}%>
 <%
-    NotifdConfigFactory.init(); //Must do this early on - if it fails, then just throw the exception to the web gui
+    NotifdConfigFactory.init();
+    NotifdConfigFactory.getInstance().reload(); //Must do this early on - if it fails, then just throw the exception to the web gui
```
Comment (freddy61025) [inline opennms-webapp/src/main/webapp/admin/sched-outages/editoutage.jsp:162]: is that possible to not call reload here ? (your init reload is removed, and postConstruct did not work properly?)

File: opennms-webapp/src/main/webapp/admin/index.jsp [hunk 4]
```diff
@@ -176,6 +176,7 @@
             <%String status = "Unknown";
                 try {
                     NotifdConfigFactory.init();
+                    NotifdConfigFactory.getInstance().reload();
```
Comment (freddy61025) [inline opennms-webapp/src/main/webapp/admin/index.jsp:179]: is that possible to not call reload here ? (your init reload is removed, and postConstruct did not work properly?)

File: opennms-webapp/src/main/java/org/opennms/web/admin/notification/noticeWizard/NotificationWizardServlet.java [hunk 5]
```diff
@@ -122,6 +122,7 @@ public void doPost(final HttpServletRequest request, final HttpServletResponse r
          */
         try {
             NotifdConfigFactory.init();
+            NotifdConfigFactory.getInstance().reload();
```
Comment (freddy61025) [inline opennms-webapp/src/main/java/org/opennms/web/admin/notification/noticeWizard/NotificationWizardServlet.java:125]: is that possible to not call reload here ? (your init reload is removed, and postConstruct did not work properly?)

**Review by freddy61025 — COMMENTED** [review]
Is it possible to add the reload back to init ? It is kind of strange to add right after init NotifdConfigFactory.getInstance().reload(); but cannot call inside init

**Comment by uguggilam** [comment]
Duplicate PR. Closing this request. Actual PR and review comments are provided in https://github.com/OpenNMS/opennms/pull/3862

**Pull request closed without merging**